ArduPlane/commands_logic.cpp

```diff
@@ -499,12 +499,8 @@ void Plane::do_loiter_to_alt(const AP_Mission::Mission_Command& cmd)
     set_next_WP(loc);
     loiter_set_direction_wp(cmd);
 
-    // used to signify primary turns goal not yet met when non-zero
-    condition_value = next_WP_loc.alt;
-    if (condition_value == 0) {
-        // the value of 0 is used to signify it has been reached. Lets bump alt to 1 which is 10cm. Close enough!
-        condition_value = 1;
-    }
+    // condition_value == 1 means alt is ignored: we've either reached it or it's not valid
+    condition_value = (next_WP_loc.alt == 0);
```

Comment:
The only reason to bump it to 1 was so that we could loiter down to 0 meters. Now we should always be just setting the condition_value to 1 as a flag that we haven't yet achieved the altitude.